src/common/transformations/include/transformations/rt_info/original_precision_attribute.hpp

```diff
@@ -0,0 +1,33 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "openvino/core/node.hpp"
+#include "openvino/core/rtti.hpp"
+#include "openvino/core/runtime_attribute.hpp"
+#include "transformations_visibility.hpp"
+
+namespace ov {
+
+TRANSFORMATIONS_API void set_original_precision_attribute(const std::shared_ptr<Node>& node,
+                                                          const element::Type_t original_precision);
+TRANSFORMATIONS_API void reset_original_precision_attribute(const std::shared_ptr<Node>& node);
+TRANSFORMATIONS_API element::Type_t get_original_precision(const std::shared_ptr<Node>& node);
+
+/**
+ * @ingroup ie_runtime_attr_api
+ * @brief OriginalPrecisionAttribute stores the original precision of the node to pass this information to
+ * plugins.
+ */
+class TRANSFORMATIONS_API OriginalPrecisionAttribute : public RuntimeAttribute {
+public:
+    OPENVINO_RTTI("original_precision", "0");
+
+    bool is_copyable() const override {
+        return false;
```

Comment:
Why? If ReadValue/Assign op is replaced with some custom one which handles state internally, then it would be good to have this attribute in the

Reply:
fixed